File: hms-sdk-unity/HuaweiMobileServices/Scan/OnLightVisibleCallBack/OnLightVisibleCallBackWrapper.cs
namespace HuaweiMobileServices.Scan
{
    using HuaweiMobileServices.Utils;

    using UnityEngine;

    // Wrapper for com.huawei.hms.hmsscankit.OnLightVisibleCallBack
    internal class OnLightVisibleCallBackWrapper : JavaObjectWrapper
    {
        private class OnLightVisibleCallBackInterfaceWrapper : AndroidJavaProxy
        {

            private readonly IOnLightVisibleCallBack mListener;

            public OnLightVisibleCallBackInterfaceWrapper(IOnLightVisibleCallBack listener) : base("org.m0skit0.android.hms.unity.scan.OnLightVisibleCallBackWrapper")
            {
                mListener = listener;
            }

            public void OnVisibleChanged(bool visible)
            {
                this.CallOnMainThread(() => { mListener.OnVisibleChanged(visible); });
            }
        }

        public OnLightVisibleCallBackWrapper(IOnLightVisibleCallBack listener)
            : base("org.m0skit0.android.hms.unity.scan.OnLightVisibleCallBackWrapper", new OnLightVisibleCallBackInterfaceWrapper(listener)) { }
    }
}

File: android-wrapper/app/src/main/java/org/m0skit0/android/hms/unity/scan/OnLightVisibleCallBackWrapper.java
package org.m0skit0.android.hms.unity.scan;

public class OnLightVisibleCallBackWrapper implements com.huawei.hms.hmsscankit.OnLightVisibleCallBack {

    private final OnLightVisibleCallBack listener;

    public OnLightVisibleCallBackWrapper(final OnLightVisibleCallBack listener) {
        super();
        this.listener = listener;
    }

    @Override
    public void onVisibleChanged(boolean b) {
        listener.onVisibleChanged(b);
    }
}
